docs/notebooks/active_learning_for_binary_classification.pct.py

```diff
@@ -0,0 +1,191 @@
+
+# %%
+numSamples = 10
+X = search_space.sample(numSamples)
+observer = mk_observer(circle, OBJECTIVE)
+datasets = observer(X)
+
+# %% [markdown]
+# ## Modelling the binary classification task
+
+# %% [markdown]
+# For the binary classification model, we use the Variational Gaussian Process with Bernoulli likelihood. For more detail of this model, see <cite data-cite="Nickisch08a">[Nickisch et al.](https://www.jmlr.org/papers/volume9/nickisch08a/nickisch08a.pdf)</cite>.
+
+# %%
+from trieste.models.gpflow import GPflowModelConfig, VariationalGaussianProcess
+
+
+def create_bo_model(data):
+    kernel = gpflow.kernels.SquaredExponential(lengthscales=[0.2, 0.2])
+    m = gpflow.models.VGP(data.astuple(), likelihood=gpflow.likelihoods.Bernoulli(), kernel=kernel)
+    return trieste.models.create_model(
+        GPflowModelConfig(
+            **{
+                "model": m,
+                "optimizer": gpflow.optimizers.Scipy(),
+                "optimizer_args": {
+                    "minimize_args": {"options": dict(maxiter=100.0)},
+                },
+            }
+        )
+    )
+
+
+"""
+# model seems harder to train using default VariationalGaussianProcess() optimiser which is adam?
+
+def create_bo_model(data):
+    kernel = gpflow.kernels.SquaredExponential()
+    m = gpflow.models.VGP(data.astuple(), likelihood=gpflow.likelihoods.Bernoulli(), kernel=kernel)
+    return VariationalGaussianProcess(m)
+"""
+
+# %% [markdown]
+# Lets see our model landscape using only those initial data
+
+# %%
+model = create_bo_model(datasets[OBJECTIVE])
+
+model.update(datasets[OBJECTIVE])
+model.optimize(datasets[OBJECTIVE])
+
+mean, variance = model.predict(xx)
+
+plt.figure()
+plt.contourf(*grid_xx, np.reshape(mean, [density] * input_dim))
+plt.plot(
+    datasets[OBJECTIVE].query_points[:, 0],
+    datasets[OBJECTIVE].query_points[:, 1],
+    "ko",
+    markersize=10,
+)
+plt.title("Mean")
+plt.colorbar()
+plt.show()
+
+plt.figure()
+plt.contourf(*grid_xx, np.reshape(variance, [density] * input_dim))
+plt.colorbar()
+plt.plot(
+    datasets[OBJECTIVE].query_points[:, 0],
+    datasets[OBJECTIVE].query_points[:, 1],
+    "ko",
+    markersize=10,
+)
+plt.title("Variance")
+plt.show()
+
+
+# %% [markdown]
+# ## The acquisition process
+#
+# We can construct the BALD acqusition function which maximise information gain about the model parameters, by maximising the mutual information between predictions and model posterior:
+#
+# $$\mathbb{I}\left[y, \boldsymbol{\theta} \mid \mathbf{x}, \mathcal{D}\right]=\mathbb{H}\left[y \mid \mathbf{x}, \mathcal{D}\right]-\mathbb{E}_{p\left(\boldsymbol{\theta} \mid \mathcal{D}\right)}[\mathbb{H}[y \mid \mathbf{x}, \boldsymbol{\theta}]]$$
+#
+# See <cite data-cite="houlsby2011bayesian">[Houlsby et al.](https://arxiv.org/pdf/1112.5745.pdf)</cite> for more details. Then, Trieste's `EfficientGlobalOptimization` is used for the query rule:
+
+# %%
+initial_models = trieste.utils.map_values(create_bo_model, datasets)
```

Comment:
you have a single dataset and model, so no need for this code

```suggestion
initial_models = create_bo_model(datasets)
```